tracing-subscriber/src/fmt/fmt_layer.rs

```diff
@@ -0,0 +1,532 @@
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Sets the layer being built to use a [less verbose formatter](../fmt/format/struct.Compact.html).
+    pub fn compact(self) -> LayerBuilder<S, N, format::Format<format::Compact, T>, W>
+    where
+        N: for<'writer> FormatFields<'writer> + 'static,
+    {
+        LayerBuilder {
+            fmt_event: self.fmt_event.compact(),
+            fmt_fields: self.fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+
+    /// Sets the layer being built to use a [JSON formatter](../fmt/format/struct.Json.html).
+    #[cfg(feature = "json")]
+    pub fn json(self) -> LayerBuilder<S, format::JsonFields, format::Format<format::Json, T>, W> {
+        LayerBuilder {
+            fmt_event: self.fmt_event.json(),
+            fmt_fields: format::JsonFields::new(),
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> LayerBuilder<S, N, E, W> {
+    /// Sets the field formatter that the layer being built will use to record
+    /// fields.
+    pub fn fmt_fields<N2>(self, fmt_fields: N2) -> LayerBuilder<S, N2, E, W>
+    where
+        N2: for<'writer> FormatFields<'writer> + 'static,
+    {
+        LayerBuilder {
+            fmt_event: self.fmt_event,
+            fmt_fields,
+            make_writer: self.make_writer,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S, N, E, W> LayerBuilder<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    /// Builds a [`Layer`] with the provided configuration.
+    ///
+    /// [`Layer`]: struct.Layer.html
+    pub fn finish(self) -> Layer<S, N, E, W> {
+        Layer {
+            make_writer: self.make_writer,
+            fmt_fields: self.fmt_fields,
+            fmt_event: self.fmt_event,
+            _inner: self._inner,
+        }
+    }
+}
+
+impl<S> Default for Layer<S>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn default() -> Self {
+        LayerBuilder::default().finish()
+    }
+}
+
+impl<S> Default for LayerBuilder<S> {
+    fn default() -> Self {
+        LayerBuilder {
+            fmt_fields: format::DefaultFields::default(),
+            fmt_event: format::Format::default(),
+            make_writer: io::stdout,
+            _inner: PhantomData,
+        }
+    }
+}
+
+impl<S, N, E, W> Layer<S, N, E, W>
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+    N: for<'writer> FormatFields<'writer> + 'static,
+    E: FormatEvent<S, N> + 'static,
+    W: MakeWriter + 'static,
+{
+    #[inline]
+    fn make_ctx<'a>(&'a self, ctx: Context<'a, S>) -> FmtContext<'a, S, N> {
+        FmtContext {
+            ctx,
+            fmt_fields: &self.fmt_fields,
+        }
+    }
+}
```

Comment:
i think this could be in the same impl block as `finish` (note that the bounds are the same)

Reply:
I don't think they can: `finish` is a method on `fmt::LayerBuilder` while `make_ctx` is a method on `fmt::Layer`.

Reply:
oh whoops, i can't read. nevermind, carry on.